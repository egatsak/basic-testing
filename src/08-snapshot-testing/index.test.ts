// Uncomment the code below and write your tests
import { generateLinkedList } from './index';

describe('generateLinkedList', () => {
  const values1 = [
    { id: 1, baz: 'bar' },
    { id: 2, baz: 'baz' },
    { id: 3, baz: 'foo' },
  ];

  const values2 = [
    { id: 11, baz: 'boo' },
    { id: 22, baz: 'damn' },
    { id: 33, baz: 'ffuuu-----' },
  ];

  const expected = {
    value: { id: 1, baz: 'bar' },
    next: {
      value: { id: 2, baz: 'baz' },
      next: {
        value: { id: 3, baz: 'foo' },
        next: {
          value: null,
          next: null,
        },
      },
    },
  };

  // Check match by expect(...).toStrictEqual(...)
  test('should generate linked list from values 1', (): void => {
    expect(generateLinkedList(values1)).toStrictEqual(expected);
  });

  // Check match by comparison with snapshot
  test('should generate linked list from values 2', (): void => {
    expect(generateLinkedList(values2)).toMatchSnapshot();
  });
});
