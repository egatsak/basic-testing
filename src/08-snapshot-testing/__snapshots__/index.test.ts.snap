// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`generateLinkedList should generate linked list from values 2 1`] = `
{
  "next": {
    "next": {
      "next": {
        "next": null,
        "value": null,
      },
      "value": {
        "baz": "ffuuu-----",
        "id": 33,
      },
    },
    "value": {
      "baz": "damn",
      "id": 22,
    },
  },
  "value": {
    "baz": "boo",
    "id": 11,
  },
}
`;
